micronota/db/uniref.py

```diff
@@ -0,0 +1,380 @@
+publicly available protein sequences in the world.
+
+The UniProt Reference Clusters (UniRef) [#]_ provide clustered sets (UniRef100,
+UniRef90 and UniRef50 clusters) of sequences from the UniProtKB
+and selected UniParc records, in order to obtain complete coverage of sequence
+space at several resolutions (100%, >90% and >50%) while hiding redundant
+sequences (but not their descriptions) from view.
+
+If you used this database through micronota, you should also `cite UniProt
+<http://www.uniprot.org/help/publications>`_ properly.
+
+Release
+-------
+:version: 2016_01
+:date:    01/20/2016
+:link:    ftp://ftp.uniprot.org/pub/databases/uniprot/relnotes.txt
+
+Files
+-----
+* UniRef files
+
+  UniRef fasta files clustered at the similarity levels of 50, 90 or 100.
+  The UniRef100 identifier is generated by placing ``UniRef100_`` prefix
+  before the UniProtKB accession number or UniParc identifier of the
+  representative UniProtKB or UniParc entry. It is similarly done for
+  UniRef50 or UniRef90.
+
+  * uniref50.fasta.gz [#]_
+
+  * uniref90.fasta.gz [#]_
+
+  * uniref100.fasta.gz [#]_
+
+* UniProtKB files
+
+  The .dat.gz files contain the UniProtKB records in a variant format of EMBL.
+  The records are divided into taxa groups [#]_.
+
+  * uniprot_sprot.dat.gz
+
+  * uniprot_trembl.dat.gz
+
+Reference
+---------
+.. [#] http://www.uniprot.org/help/uniparc
+.. [#] http://www.uniprot.org/help/uniprotkb
+.. [#] http://www.ncbi.nlm.nih.gov/pubmed/17379688
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref50/README
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref90/README
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref100/README
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/taxonomic_divisions/README
+'''
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, micronota development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from os.path import join, basename
+from sqlite3 import connect
+from xml.etree import ElementTree as ET
+import gzip
+
+from skbio import read, Sequence
+
+from ..util import _overwrite, _download
+
+
+def prepare_db(out_d, downloaded, force=False,
+               sprot='ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.xml.gz',
+               trembl='ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_trembl.xml.gz',
+               uniref100='ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref100/uniref100.fasta.gz',
+               id_map='ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/idmapping_selected.tab.gz'):
```

Comment:
sure. we can think about it in the future.

On Wed, Feb 17, 2016 at 2:09 PM, Jamie Morton <notifications@github.com>
wrote:

> In micronota/db/uniref.py
> <https://github.com/biocore/micronota/pull/43#discussion_r53239387>:
>
> > +
> > +from os.path import join, basename
> > +from sqlite3 import connect
> > +from xml.etree import ElementTree as ET
> > +import gzip
> > +
> > +from skbio import read, Sequence
> > +
> > +from ..util import _overwrite, _download
> > +
> > +
> > +def prepare_db(out_d, downloaded, force=False,
> > +               sprot='ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.xml.gz',
> > +               trembl='ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_trembl.xml.gz',
> > +               uniref100='ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref100/uniref100.fasta.gz',
> > +               id_map='ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/idmapping_selected.tab.gz'):
>
> gawd this is hideous. We need to discuss potentially moving these
> variables into a separate config file in the future.ew
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/micronota/pull/43/files#r53239387>.
>